next-client/src/server-env.ts

```diff
@@ -84,3 +57,90 @@ if (!result.success) {
  * NextJS server env vars. Next server should fail to start if not present.
  */
 export const serverEnv = result.data;
+
+/**
+ * Client environment validation schema
+ */
+export const clientEnv = z.object({
+  PIPELINE_EXPRESS_URL: z.string(),
+  NEXT_PUBLIC_FIREBASE_API_KEY: z.string(),
+  NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN: z.string(),
+  NEXT_PUBLIC_FIREBASE_PROJECT_ID: z.string(),
+  NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET: z.string(),
+  NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID: z.string(),
+  NEXT_PUBLIC_FIREBASE_APP_ID: z.string(),
+  NODE_ENV: z.string().optional().default("development"),
+});
+
+export type ClientEnv = z.infer<typeof clientEnv>;
+
+/**
+ * Parse and validate client environment variables
+ */
+export function validateClientEnv(): ClientEnv {
+  const parsed = clientEnv.safeParse(process.env);
+
+  if (!parsed.success) {
```

Comment:
when I was working on this branch, I needed this change for types to be correct:
- if (!parsed.success) {
+ if (parsed.success === false) {